enforced double precision calculations in *Group.center()

zemanj opened the pull request:
Fixes #1930 (at least in part)

Changes made in this Pull Request:
 - enforced calculations in `*Group.center()` to be performed in double precision
 - for compatibility reasons, added automatic type conversion to `dtype=np.float32` for input coordinate arrays of `lib.distances.apply_PBC()`, `lib.distances.distance_array()`, and `lib.distances.self_distance_array()`
 - arrays returned by `*Group.center()` will be of `dtype=np.float64` unless ``(compound != 'group' and pbc==True)``. This exception could be eliminated by providing a double-precision version of `lib.distances.apply_PBC()`.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

richardjgowers commented:
@zemanj if you're doing a few you may as well do all the functions in distances (eg `calc_bonds`)

zemanj commented:
@richardjgowers I only did the ones that were needed to pass the tests since I didn't know whether you guys are ok with what I'm doing. Sure I can do that for the remaining functions as well.
Updates for tests, docs, and changelog are also needed.

orbeckst commented:
@richardjgowers did you already take this PR into account in your PR #1939? 

If not, should we merge this one "as is" and then worry about double precision in your PR (rebased) or do a new one to add remaining functionality?

zemanj commented:
Recent changes:
- Added typecasting of input coordinate arrays to `np.float32` for all accessible functions in `lib.distances`
- Changed docstrings accordingly
- Removed respective "wrong dtype" tests, parametrized tests for `np.float32` and `np.float64` instead.

Open questions:
- Do we need tests for other dtypes?

Will add CHANGELOG entry once this passes.

richardjgowers commented:
@zemanj I think float32 & 64 is enough
